src/server.c

```diff
@@ -3549,7 +3555,7 @@ void afterCommand(client *c) {
          * Should be done before trackingHandlePendingKeyInvalidations so that we
          * reply to client before invalidating cache (makes more sense) */
         if (server.core_propagates)
-            propagatePendingCommands();
+            afterDatasetChange();
```

Comment:
another case where i think it's wrong to condition afterDatasetChange on a propagation related variable.
maybe it can be moved to inside afterDatasetChange if needed.

Reply:
As I wrote here: https://github.com/redis/redis/pull/11199#discussion_r979391064
I prefer to do it on separate PR if you agree.